spec/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser_spec.rb

```diff
@@ -36,16 +36,68 @@
     expect(described_class.miq_template_type).to eq("ManageIQ::Providers::Lenovo::PhysicalInfraManager::Template")
   end
 
+  context 'parse physical chassis' do
+    let(:result) do
+      VCR.use_cassette("#{described_class.name.underscore}_ems_inv_to_hashes") do
+        ems_inv_to_hashes
+      end
+    end
+
+    it 'will retrieve physical all chassis' do
+      expect(result[:physical_chassis].size).to eq(1)
+    end
+
+    it 'will parse physical chassi fields' do
+      physical_chassi = result[:physical_chassis].first
+
+      expect(physical_chassi[:name]).to eq("SN#Y034BG16E02C")
+      expect(physical_chassi[:uid_ems]).to eq("9841028B07714FD09D9297C6D1A4943E")
+      expect(physical_chassi[:ems_ref]).to eq("9841028B07714FD09D9297C6D1A4943E")
+      expect(physical_chassi[:overall_health_state]).to eq("Warning")
+      expect(physical_chassi[:management_module_slot_count]).to eq(2)
+      expect(physical_chassi[:switch_slot_count]).to eq(4)
+      expect(physical_chassi[:fan_slot_count]).to eq(10)
+      expect(physical_chassi[:blade_slot_count]).to eq(14)
+      expect(physical_chassi[:powersupply_slot_count]).to eq(6)
+      expect(physical_chassi[:health_state]).to eq("Valid")
+      expected_type = "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalChassis"
+      expect(physical_chassi[:type]).to eq(expected_type)
+    end
+
+    it 'will retrieve physical chassis asset detail data' do
+      asset_detail = result[:physical_chassis].first[:asset_detail]
+      expect(asset_detail[:product_name]).to eq("IBM Chassis Midplane")
+      expect(asset_detail[:manufacturer]).to eq("IBM")
+      expect(asset_detail[:machine_type]).to eq("7893")
+      expect(asset_detail[:model]).to eq("92X")
+      expect(asset_detail[:serial_number]).to eq("100080A")
+      expect(asset_detail[:contact]).to eq("No Contact Configured")
+      expect(asset_detail[:description]).to eq("Lenovo Flex System Chassis")
+      expect(asset_detail[:location]).to eq("No Location Configured")
+      expect(asset_detail[:room]).to eq("")
+      expect(asset_detail[:rack_name]).to eq("")
+      expect(asset_detail[:lowest_rack_unit]).to eq(0)
+    end
+
+    it 'will have a reference to the physical rack where the chassis is inside' do
+      physical_rack = result[:physical_racks].first
+      physical_chassis = result[:physical_chassis].first
+
+      expect(physical_chassis[:physical_rack][:ems_ref]).to eq(physical_rack[:ems_ref])
```

Comment:
Seems like this is a redundant test if you are verifying the equality between `physical_chassis[:physical_rack]` and `physical_rack[:ems_ref]` as you do bellow.